### PR TITLE
Conditionally load droplet/ace - only when not in hideSource mode

### PR DESCRIPTION
**Pull request by cpirich**
We've noticed for some time that we unnecessarily load a bunch of script code in the "share" experience - when viewing someone else's creation (but not in the case where you click "How it works"). For droplet/editCode, we now avoid loading droplet, ace, and associated CSS when we know the editor will not appear. This is the case when `hideSource` is set, true in the aforementioned share scenario and also the "embed" mode.
- Created `StudioApp.prototype.getCode` as way to return the code that should be used for applab/playlab while in editCode mode. Replaced most uses of `studioApp.editor.getValue()` with calls to this function. `Applab.getCode()` still exists as a wrapper since the dashboard/shared code expects to find it. For cases where the editor is not present, we return a copy of the initial code (stored away in `this.startBlocks_`)
- Modified `JSInterpreter` to avoid building its `codeInfo` structure when in `hideSource` mode. Similarly, early exit on the `selectCurrentCode`, `getUserCodeLine`, and `getNearestUserCodeLine` methods.
- Moved `Applab.startSharedAppAfterWarnings` to run in `afterInject` instead of `afterEditorReady`. `afterEditorReady` will not fire when there is no editor in `hideSource` mode.
- Modified dashboard's `render_app_dependencies` to leverage `hide_source` and `use_applab` to more appropriately load the script code that will be needed.

### CODE REVIEW

**Comment by cpirich** [comment]
@joshlory @islemaster

**Comment by Bjvanminnen** [comment]
Would be great if we could have some sort of test coverage of this. Not sure how easy this would be to do at the mochaTest level since it depends on hideSource, which is on appOptions rather than on appOptions.level. Maybe we can just create a simple UI test though that loads a share page (actually we already have on in applab/sharedApps.feature) and then does something to validate that droplet wasn't loaded.

Not sure exactly what that something is, but I suspect there are some globals that we set in the case where droplet is loaded that won't be set now?

File: dashboard/test/ui/features/applab/sharedApps.feature [hunk 1]
```diff
@@ -18,6 +18,8 @@ Feature: App Lab Scenarios
 
     When I navigate to the shared version of my project
     And I wait until element "#divApplab > .screen > button#hello" is visible
+    Then evaluate JavaScript expression "typeof window.ace === 'undefined'"
+    Then evaluate JavaScript expression "typeof window.droplet === 'undefined'"
```
Comment (islemaster) [inline dashboard/test/ui/features/applab/sharedApps.feature:22]: :+1: Have you run these against our supported browsers?

**Comment by islemaster** [comment]
Awesome Chris!  Do you have any numbers on load time / download size savings?

**Comment by cpirich** [comment]
Looks like about 500KB of script code that we will no longer load in this case - spread across 10-12 separate requests. I haven't tried to compare the timing.

**Pull request merged**